src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -48,6 +48,11 @@ internal AssemblyScanner(Assembly assemblyToScan)
         /// </summary>
         public bool ScanAppDomainAssemblies { get; set; } = true;
 
+        /// <summary>
+        /// Determines if the scanner should scan assemblies from the application directory.
+        /// </summary>
+        public bool ScanFileSystemAssemblies { get; set; } = true;
```

Comment:
Would it be better to rename this to something like

`ScanApplicationBaseDirectoryAssemblies`?

Reply:
hm maybe that would make sense for the user facing setting on the EndpointConfiguration, but here you can control the directory via ctor params? (yes it's kinda weird that you can configure a directory and then disable it, but I planned to address that via some refactoring via later PRs so we don't have to backport the full refactoring)

Reply:
I meant is more as a larger comment to rename the property on the whole hierarchy because right now they are called the same so I assumed making the comment here only is good enough

Reply:
thinking a bit more about this, this setting also can scan sub-directories (`ScanAssembliesInNestedDirectories` setting) and even totally different paths (`AdditionalAssemblyScanningPath`), so naming it after AppBaseDirectory seems to be slightly incorrect?

Reply:
I've kept the name of the option as currently but improved the documentation to give a better understanding of the behavior on the user-facing API here: https://github.com/Particular/NServiceBus/pull/6239/files#diff-31d7176930fa411f75a34ecb62a45abf96c3d9908cbda5e198acb669f501ca01R20
(note that the doc on this property is more minimal as the AssemblyScanner has the related settings internal so they can't be mentioned here)